### PR TITLE
feat: cortex update command

### PR DESCRIPTION
**Pull request by vansangpfiev**
https://github.com/janhq/cortex.cpp/issues/1087 
Only support to download latest version for now
Also added `cortex -v` command
https://github.com/janhq/cortex.cpp/issues/1066
```
.\cortex.exe -v
v0.5.0
```

### CODE REVIEW

**Comment by namchuai** [comment]
Nice one! I have one question. What happen in windows, if the server is running, then our executable will be in used, right? ? will `std::rename` fail and failed the update, I think?

File: engine/commands/cortex_upd_cmd.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#include <optional>
+
+namespace commands {
+
+class CortexUpdCmd{
+ public:
+  CortexUpdCmd();
+  void Exec(std::string version);
+
+ private:
```
Comment (namchuai) [inline engine/commands/cortex_upd_cmd.h]: small nit: maybe remove `private` if we don't have any field

File: engine/commands/cortex_upd_cmd.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  } else {
+    auto err = res.error();
+    CTL_ERR("HTTP error: " << httplib::to_string(err));
+    return;
+  }
+#if defined(_WIN32)
+  std::string temp = ".\\cortex_tmp.exe";
+  remove(temp.c_str());  // ignore return code
+
+  std::string src =
```
Comment (namchuai) [inline engine/commands/cortex_upd_cmd.cc]: small nit, apply `std::filesystem::path` for concatenation path will simplify the code.

File: engine/commands/cortex_upd_cmd.cc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      ".\\cortex\\" + kCortexBinary + "\\" + kCortexBinary + ".exe";
+  std::string dst = ".\\" + kCortexBinary + ".exe";
```
Comment (namchuai) [inline engine/commands/cortex_upd_cmd.cc]: I think using `file_manager_utils::GetExecutableFolderContainerPath` might be better for storing files relatively to our binary (executable)

**Review by namchuai — APPROVED** [review]
✅ Looks good. Just some small nits. Not required though.

**Review by nguyenhoangthuan99 — APPROVED** [review]

**Comment by vansangpfiev** [comment]
> Nice one! I have one question. What happen in windows, if the server is running, then our executable will be in used, right? ? will `std::rename` fail and failed the update, I think?

We cannot replace but can remove file on windows if it is in used.
Also added a TODO to stop server if it is running before updating.

**Comment by dan-menlo** [comment]
I am removing this from Project Board:

- Project Board is for issues, issue is alr tracked in #1087 
- Issues should be closed via PR, via close keyword

**Pull request merged**